src/main/frontend/client/components/TrailEdit.js

```diff
@@ -0,0 +1,144 @@
+import React, { useState, useEffect } from 'react';
+import { Redirect } from 'react-router-dom';
+
+import ErrorList from './ErrorList';
+import DifficultyFormField from './DifficultyFormField';
+
+const TrailEdit = (props) => {
+  const trailId = props.match.params.id;
+  const {name, description, distance, elevationGain, difficulty, zipCode, imgUrl} = props.location.state.trail;
+  const [errors, setErrors] = useState({});
+  const [shouldRedirect, setShouldRedirect] = useState(false);
+  const [formData, setFormData] = useState({
+     id: trailId,
+     name: name,
+     description: description,
+     distance: distance,
+     elevationGain: elevationGain,
+     difficulty: difficulty,
+     zipCode: zipCode,
+     imgUrl: imgUrl
+  });
+
+
+  const updateTrail = async () => {
+    try {
+      const res = await fetch(`/api/v1/trails/${trailId}/edit`, {
+        method: 'PUT',
+        headers: new Headers({
+          'Content-Type': 'application/json'
+        }),
+        body: JSON.stringify(formData)
+      });
+      if (!res.ok) {
+        if(res.status === 422) {
+          const body = await res.json();
+          return setErrors(body.errors);
+        } else {
+          const error = new Error(`${res.status} (${res.statusText})`);
+          throw(error);
+        }
+      }
+      setShouldRedirect(true);
+    } catch(err) {
+      console.error(`Error in fetch: ${err.message}`)
+    }
+  };
+
+  const changeHandler = event => {
+    setFormData({
+      ...formData,
+      [event.currentTarget.name]: event.currentTarget.value
+    });
+  };
+
+  const submitFormHandler = event => {
+    event.preventDefault();
+    setErrors({});
+    updateTrail();
+  };
+
+  if (shouldRedirect) {
+    return <Redirect push to={`/trails/${trailId}`} />;
+  }
+
```

Comment:
This all looks great when a user uses the link to navigate here.  What happens if i type in `/trails/1/edit` and press enter?  Will the form be pre populated with the correct trail information?  We may need to refactor this to send a GET fetch for the form information in a useEffect